lib/WorseReflection/Bridge/Phpactor/DocblockParser/PHPStanDocblockParserFactory.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Phpactor\WorseReflection\Bridge\Phpactor\DocblockParser;
+
+use Phpactor\WorseReflection\Core\DocBlock\PlainDocblock;
+use Phpactor\WorseReflection\Core\DocBlock\DocBlock;
+use Phpactor\WorseReflection\Core\DocBlock\DocBlockFactory;
+use Phpactor\WorseReflection\Core\Reflection\ReflectionScope;
+use Phpactor\WorseReflection\Reflector;
+use PHPStan\PhpDocParser\Lexer\Lexer;
+use PHPStan\PhpDocParser\Parser\ConstExprParser;
+use PHPStan\PhpDocParser\Parser\PhpDocParser;
+use PHPStan\PhpDocParser\Parser\TokenIterator;
+use PHPStan\PhpDocParser\Parser\TypeParser;
+
+class PHPStanDocblockParserFactory implements DocBlockFactory
+{
+    private PhpDocParser $parser;
+
+    public function __construct(
+        private Reflector $reflector,
+        private Lexer $lexer = new Lexer(),
+    ) {
+        $constExprParser = new ConstExprParser();
+        $typeParser = new TypeParser($constExprParser);
+        $this->parser = new PhpDocParser($typeParser, $constExprParser);
+    }
+
+    public function create(string $docblock, ReflectionScope $scope): DocBlock
+    {
+        if (trim($docblock) === '') {
+            return new PlainDocblock();
+        }
+
```

Comment:
note that you _will_ need to filter for "supported tags" by regex before proceeding to parse as with the previous factory (maybe it can be done with a decorator). Otherwise we parse every single docblock whether we extract info from it or not.

Reply:
By supported you mean those, which has according `Type` classes in worse reflection? I've ditched this, because I thought those are tags supported by doc parser

Reply:
i mean:
```
    const SUPPORTED_TAGS = [
        'property',
        'var',
        'param',
        'return',
        'method',
        'type',
        'deprecated',
        'extends',
        'implements',
        'template',
        'template-covariant',
        'template-extends',
        'mixin',
        'throws',
        'assert',
    ];
```
are the types that are analysed, if there is a docblock that has none of those then we don't need to parse it.

Reply:
added